File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head == NULL || head->next == NULL) return head;

        ListNode* dh = new ListNode(-1);
        ListNode* dt = dh;

        ListNode* prev = NULL;
        ListNode* cur = head;

        while(cur != NULL){

            if(prev == NULL || prev->val != cur->val){
                dt->next = cur;
                dt = dt->next;

                prev = cur;
               
            }
            
            cur = cur->next;
        }

        dt->next = NULL;
        return dh->next;
    }
};

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        
        if(head == null || head.next == null) return head;
        
        ListNode prev = null;
        ListNode cur = head;
        
        ListNode dh = new ListNode(-1);
        ListNode dt = dh;
        
        while(cur != null){
            
            if(prev == null || prev.val != cur.val){
                dt.next = cur;
                dt = dt.next;
                
                prev = cur;
                cur = cur.next;
            }
            else cur = cur.next;
        }
        
        dt.next = null;
        return dh.next;
    }
}